hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -113,6 +115,7 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
 
         for (final var allowance : tokenAllowances) {
             validateNullableAccountID(allowance.owner());
+            validateFalsePreCheck(allowance.amount() > 0 && allowance.spender() == null, INVALID_ALLOWANCE_SPENDER_ID);
```

Comment:
We should probably refactor to pass the response code `INVALID_ALLOWANCE_SPENDER_ID)` into `validateAccountID` on line 119. It is already doing the same check imo. This line is not needed.

Reply:
allowance.amount() was not checked in `validateAccountID`. Should I also include this check on line 112 too as I saw similar check in mono [here](https://github.com/hashgraph/hedera-services/blob/8aeb2d68d61574e74f77b728211a8001ab38f8cb/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java#L113).

Reply:
Updated the change.